### PR TITLE
[aws-saml-idp] integration to manage SAML IDPs

### PR DESCRIPTION
**Pull request by chassing**
Introduce a new integration to manage AWS accounts' SAML IDP configuration.

Ticket: [SDE-3516](https://issues.redhat.com/browse/SDE-3516)
Design doc: [appsre-aws-account-sso](https://gitlab.cee.redhat.com/service/app-interface/-/blob/master/docs/app-sre/design-docs/appsre-aws-account-sso.md)

Additionally:
* update integration list
* and lint README

### CODE REVIEW

File: reconcile/test/aws_saml_idp/test_aws_saml_idp_integration.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    intg: AwsSamlIdpIntegration, fixture_query_func: Callable
+) -> None:
+    state = intg.get_early_exit_desired_state(query_func=fixture_query_func)
+    assert "accounts" in state
+
+
+def test_aws_saml_idp_get_aws_accounts(
+    gql_class_factory: Callable, aws_accounts: list[AWSAccountV1]
+) -> None:
+    assert aws_accounts == [
```
Comment (hemslo) [inline reconcile/test/aws_saml_idp/test_aws_saml_idp_integration.py:19]: Only assert in test is a little confusing, a clear test structure is `Arrange-Act-Assert`, arrange can be done via fixtures, but if `Act` is also hidden in fixture, it's hard to tell what case is being tested here.

Comment (chassing) [reply reconcile/test/aws_saml_idp/test_aws_saml_idp_integration.py]: Fixed in [707c9d5](https://github.com/app-sre/qontract-reconcile/pull/4060/commits/707c9d537869af8e9c7cb6adb734a36a0dff6a34)

File: reconcile/aws_saml_idp/integration.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+            QONTRACT_INTEGRATION_VERSION,
+            "",
+            aws_accounts_dict,
+            working_dirs,
+            self.params.thread_pool_size,
+            aws_api,
+        )
+
+        if tf is None:
+            sys.exit(ExitCodes.ERROR)
```
[the remark this answers is not in the document]
Comment (chassing) [reply reconcile/aws_saml_idp/integration.py]: Fixed in [b22ddf8](https://github.com/app-sre/qontract-reconcile/pull/4060/commits/b22ddf8d879014082db3dc409758f92a41fd001e)

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**